src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+* [starter](https://github.com/dbanksdesign/style-dictionary-dark-mode/tree/starter): the project before any dark mode stuff
+* [single-token](https://github.com/dbanksdesign/style-dictionary-dark-mode/tree/single-token): the full implementation using the single-token approach
+* [multi-file](https://github.com/dbanksdesign/style-dictionary-dark-mode/tree/multi-file): the full implementation using the multi-file approach with file extensions. 
+
+Before we start, let's look at how different platforms handle color modes. If you don't care about this stuff and want to just jump into the setup you can skip these sections.
+
+
+----
+
+
+## Web
+
+The built-in way to support color modes on the web is to use the [`prefers-color-scheme` CSS media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme). The available options for this query are: light, dark, and no-preference.
+
+```css
+@media (prefers-color-scheme: dark) {
+  /* Dark mode styles here */
+}
+
+@media (prefers-color-scheme: light) {
+  /* Light mode styles here */
+}
+```
+
+You can conditionally load CSS files based on media query:
+
+```html
+<!-- No media query, will always be loaded -->
+<link rel="stylesheet" href="/base.css" />
+<!-- Will only be loaded if user prefers light color scheme or has no preference -->
+<link rel="stylesheet" href="/light.css" media="(prefers-color-scheme: light) or (prefers-color-scheme: no-preference)" />
+<!-- Will only be loaded if the user prefers a dark color scheme -->
+<link rel="stylesheet" href="/dark.css" media="(prefers-color-scheme: dark)" />
+```
+
+You can use the media query to show different images:
+
+```html
+<picture>
+  <!-- Use this image if the user's OS setting is light or unset -->
+  <source srcset="photo-light.png" media="(prefers-color-scheme: light) or (prefers-color-scheme: no-preference)">
+  <!-- Use this image if the user's OS setting is dark -->
+  <source srcset="photo-dark.png" media="(prefers-color-scheme: dark)">
+  <img src="photo-light.png" />
+</picture>
+```
+
+You can use the media query in Javascript too:
+
+```javascript
+if (window.matchMedia && window.matchMedia(`(prefers-color-scheme: dark)`).matches) {
+  // dark mode
+} else {
+  // light mode
+}
+
+// You can add an event listener to do something when the media query changes
+// because a user changed their preference
+var darkQuery = window.matchMedia('(prefers-color-scheme: dark)');
+darkQuery.addListener(function(e) {
+  var newTheme = e.matches ? 'dark' : 'light';
+  // do something here
+});
+```
+
+If you wanted, you could write completely separate sets of styles for light and dark mode for each component. But that sounds a bit tedious and error prone. Instead, people tend to use [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) (sometimes called CSS variables) in addition to the `prefers-color-scheme` media query:
```

Comment:
Technical comment:
people tend to use [CSS custom properties] > in my opinion it also depends on the targeted browser compability.